mycroft/client/speech/mic.py

```diff
@@ -647,7 +648,10 @@ def listen(self, source, emitter, stream=None):
         if self.save_utterances:
             LOG.info("Recording utterance")
             stamp = str(datetime.datetime.now())
-            filename = "/tmp/mycroft_utterance%s.wav" % stamp
+            filename = "/{}/mycroft_utterances/{}.wav".format(
```

Comment:
This adds a new directory (`'save_dir'/mycroft_utterances`) and doesn't ensure that the directory exists. So currently the voice client will error out here unless the folder is manually.

Perhaps in the init there should be something like
```python
        self.saved_utterances_path = join(self.save_path, 'mycroft-utterances')
        if not isdir(self.saved_wake_words_dir):                
            os.mkdir(self.saved_wake_words_dir)
```

Reply:
Good point, added that and did the same for the `wake_words_dir`, also made the `save_wake_words` flag explicitly `False` if unspecified. If the `save_*` flags are unspecified or `False`, the directories are not created.